File: 백준/Silver/19941. 햄버거 분배/햄버거 분배.py
import sys

N, K = map(int, sys.stdin.readline().strip().split())
arr = list(sys.stdin.readline().strip())

person = 0 # 먹은 사람의 수

for i in range(N):
  if arr[i] == "P": # 만약에 해당 지점이 사람이면
    # 배열의 가능한 가장 왼쪽부터 먹어야 오른쪽 사람이 남은 걸 먹음.
    for j in range(max(0, i-K), min(N, i+K+1)):
      if arr[j] == "H": # H면 먹고
        person += 1 
        arr[j] = "E" # E로 바꿔서 먹었다고 표시
        break
      
print(person)

File: 백준/Silver/19941. 햄버거 분배/햄버거 분배.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {

    static boolean[] visited;
    static int N, K;
    static char[] arr;

    public static void main(String[] args) throws IOException {
        BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));

        String[] sa = bf.readLine().split(" ");
        int N = Integer.parseInt(sa[0]);
        int K = Integer.parseInt(sa[1]);

        String s = bf.readLine();
        visited = new boolean[N];

        arr  = s.toCharArray();

        int result = 0;

        for(int i=0; i<N; i++){
            if(arr[i] == 'P'){
                for(int j=i-K; j<=i+K; j++){
                    if(j < 0 || j >= N || visited[j]) continue;
                    if(arr[j] == 'H'){
                        result++;
                        visited[j] = true;
                        break;
                    }
                }
            }
        }

        System.out.println(result);

    }
}

